tests-it/Kafka/IntegrationSpec.hs

```diff
@@ -254,3 +258,9 @@ testMessages t =
 sendMessages :: [ProducerRecord] -> KafkaProducer -> IO (Either KafkaError ())
 sendMessages msgs prod =
   Right <$> (forM_ msgs (produceMessage prod) >> flushProducer prod)
+
+sendMessagesSync :: [ProducerRecord] -> KafkaProducer -> IO (Maybe KafkaError)
```

Comment:
Also, can we make this function to be `sendMessageSync :: ProducerRecord -> KafkaProducer -> IO (Either KafkaError ())` instead?

I think `EIther` is better from the API perspective, and `[]` is not necessary since the "list" behaviour can easily be recovered at the client by traversing the list.

Another reason I don't really want `[]` in this API is that it:
1) Looks confusing. If I saw it I would totally expect that the whole batch is sent "normally", and the function releases control after that. Not that each message is sent synchronously one by one (implication: performance).
2) When something fails, I have no way to know how far did it go through the list, and it may be important.